app/code/Magento/PageCache/etc/varnish4.vcl

```diff
@@ -166,7 +166,7 @@ sub vcl_backend_response {
 
     # cache only successfully responses and 404s
     if (beresp.status != 200 && beresp.status != 404) {
-        set beresp.ttl = 0s;
+        set beresp.ttl = 120s;
```

Comment:
What is the purpose of caching 500 / 503 / 403?

Reply:
it doesn't cache them because of the following `set beresp.uncacheable = true;`. Instead Varnish will remember that these objects are uncacheable, and will disable request collapsing for them.

If you keep the TTL to 0, you get into something like this:
- 20 users request with the same object
- varnish only goes to the backend once (request coalescing)
- varnish realizes it cannot reuse the object, so it only serves it to one client
- we now have 19 users request with the same object
- varnish only goes to the backend once
- varnish realizes ....
rinse, repeat. Ultimately, the last request needs to wait 20 round-trip to the backend before being delivered.

If you set the TTL to some positive value AND the uncacheable bit to true:
- 20 users request with the same object
- varnish only goes to the backend once (request coalescing)
- varnish realizes it cannot reuse the object, so it only serves it to one client, but remembers it as uncacheable 
- the 19 remaining requests are woken up, and can all go to the backend in parallel

it's one of the trickiest Varnish gotchas, so please let me know if I wasn't clear.

Reply:
@gquintard Your explanation sounds fine for me. :heavy_check_mark:

Reply:
Shouldn't this be set in the varnish5 and varnish6 vcl too?